systemd/systemd.go

```diff
@@ -563,6 +614,51 @@ func (s *systemd) StartNoBlock(serviceNames []string) error {
 	return err
 }
 
+// Check if unit is in failed state
+func (s *systemd) IsFailed(serviceNames []string) bool {
```

Comment:
I recommend renaming this to `UnitIsFailed` or `ServiceIsFailed`, otherwise it looks like we are checking for some failure in systemd itself.

Reply:
Maybe also changing the parameter from an array to `...`? No strong preference here, but since I see lots of `...` in this function, maybe it could make things simpler?

Reply:
OK :-)